crrExistingObjects.js

```diff
@@ -0,0 +1,166 @@
+            const { Rules, Role } = repConfig;
+            const destination = Rules[0].Destination.Bucket;
+            const storageClass = Rules[0].Destination.StorageClass;
+            // set replication properties
+            const ops = objMD['content-length'] === 0 ? ['METADATA'] :
+                ['METADATA', 'DATA'];
+            const backends = [{
+                site: storageClass,
+                status: 'PENDING',
+                dataStoreVersionId: '',
+            }];
+            const replicationInfo = {
+                status: 'PENDING',
+                backends,
+                content: ops,
+                destination,
+                storageClass,
+                role: Role,
+                storageType: '',
+            };
+            objMD.replicationInfo = replicationInfo;
+            const mdBlob = JSON.stringify(objMD);
+            return bb.putMetadata({
+                Bucket: bucket,
+                Key: key,
+                ContentLength: Buffer.byteLength(mdBlob),
+                Body: mdBlob,
+            }, next);
+        },
+    ], cb);
+}
+
+// list object versions
+function _listObjectVersions(bucket, VersionIdMarker, KeyMarker, cb) {
+    return s3.listObjectVersions({
+        Bucket: bucket,
+        MaxKeys: LISTING_LIMIT,
+        VersionIdMarker,
+        KeyMarker,
+    }, cb);
+}
+
+function _markPending(bucket, versions, cb) {
+    const options = { Bucket: bucket };
+    waterfall([
+        next => s3.getBucketReplication(options, (err, res) => {
+            if (err) {
+                log.error('error getting bucket replication', { error: err });
+                return next(err);
+            }
+            return next(null, res.ReplicationConfiguration);
+        }),
+        (repConfig, next) => mapLimit(versions, 10, (i, apply) => {
+            const { Key, VersionId } = i;
+            _markObjectPending(bucket, Key, VersionId, repConfig, apply);
+        }, next),
+    ], cb);
+}
+
+function triggerCRR(bucketName, cb) {
+    const bucket = bucketName.trim();
+    let VersionIdMarker = null;
+    let KeyMarker = null;
+    doWhilst(
+        done => _listObjectVersions(bucket, VersionIdMarker, KeyMarker,
+            (err, data) => {
+                if (err) {
+                    log.error('error listing object versions', { error: err });
+                    return done(err);
+                }
+                VersionIdMarker = data.NextVersionIdMarker;
+                KeyMarker = data.NextKeyMarker;
+                return _markPending(bucket, data.Versions, done);
+            }),
+        () => {
+            if (VersionIdMarker || KeyMarker) {
+                return true;
+            }
+            return false;
+        },
+        err => {
+            if (err) {
+                log.error('error marking objects for crr', { bucket });
+                return cb(err);
+            }
+            log.info(`completed task for bucket: ${bucket}`);
```

Comment:
minor: this log could be made more consistent by logging the bucket name as an extra param `{ bucket }`